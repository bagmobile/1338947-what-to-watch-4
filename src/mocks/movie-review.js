export const mockReviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: `Kate Muir`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious
              Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in
              years.`,
    rating: 8.8
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `Bill Goodykoontz`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Anderson's films are too precious for some, but for those of us willing to lose
              ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added
              a hint of gravitas to the mix, improving the recipe.`,
    rating: 8.9
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `Matthew Lickona`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `The mannered, madcap proceedings are often delightful, occasionally silly, and
              here and there, gruesome and/or heartbreaking.`,
    rating: 3.9
  },
  {
    id: 4,
    user: {
      id: 4,
      name: `Amanda Greever`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `I didn't find it amusing, and while I can appreciate the creativity, it's an
              hour and 40 minutes I wish I could take back.`,
    rating: 5
  },
  {
    id: 5,
    user: {
      id: 5,
      name: `Paula Fleri-Soler`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 6.7
  },
  {
    id: 5,
    user: {
      id: 5,
      name: `Paula Fleri-Soler`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 8.9
  }
];
