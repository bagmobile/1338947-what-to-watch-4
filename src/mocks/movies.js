export const mockMovies = [
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Crime`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 54,
    isFavorite: false,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Crime`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 87,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 3,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Comedies`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 154,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 4,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Dramas`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 67,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
];

export const genres = [`All genres`, `Dramas`, `Comedies`, `Romance`, `Sci-Fi`, `Crime`, `Documentary`];
