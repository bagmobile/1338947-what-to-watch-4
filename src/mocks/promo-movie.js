export const PromoMovie = {
  title: `The Grand Budapest Hotel`,
  img: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  genre: `Crime`,
  year: 2010
};
