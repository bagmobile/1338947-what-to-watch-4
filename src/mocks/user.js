export const mockUserProfile = {"id": 1, "email": `test@test.ru`, "name": `test`, "avatar_url": `/wtw/static/avatar/5.jpg`};
