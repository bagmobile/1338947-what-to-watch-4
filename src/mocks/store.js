import NameSpace from "../reducer/name-space";
import {mockMovies} from "./movies";
import {mockReviews} from "./movie-review";
import {DEFAULT_GENRE, DEFAULT_MOVIE_LIST_SIZE} from "../consts";
import {AuthorizationStatus} from "../reducer/user/user";

export const testStore = {
  [NameSpace.DATA]: {
    movies: mockMovies,
    promoMovie: mockMovies[0],
    reviews: mockReviews
  },
  [NameSpace.MOVIES_LIST]: {
    activeGenre: DEFAULT_GENRE,
    currentMovieListSize: DEFAULT_MOVIE_LIST_SIZE
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  }
};
