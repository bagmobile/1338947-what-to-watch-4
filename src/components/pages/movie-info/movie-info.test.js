import React from "react";
import renderer from "react-test-renderer";
import MovieInfo from "./movie-info.connect";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {BrowserRouter, Route} from "react-router-dom";
import {testStore} from "../../../mocks/store";
import {createAPI} from "../../../api";

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);
const store = mockStore(testStore);


describe(`MovieInfo component`, () => {

  it(`Render`, () => {

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Route path={`movies/1`}>
                <MovieInfo/>
              </Route>
            </BrowserRouter>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
