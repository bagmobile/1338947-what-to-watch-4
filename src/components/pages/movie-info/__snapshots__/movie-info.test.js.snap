// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo component Render 1`] = `null`;
